### DISCUSSION TITLE
How to check for success of Import-D365Bacpac

### DISCUSSION THREAD

**Discussion by jakobschroeber** [d365collaborative/d365fo.tools#723] [Q&A] [answered]
Hi,

from the output section in the docs of the command **Import-D365Bacpac**, I couldn't find no standard response object. For other commands (like for instance **Invoke-D365AzCopyTransfer**) I have already successfully used their response objects for branching based on their values. But when I tried

```
Write-Host "$(Get-Date) - Attempting to import bacpac file"
Set-D365SqlPackagePath -Path $sqlPackagePath
$importBacpacResponse = Import-D365Bacpac -ImportModeTier1 -BacpacFile $downloadPath -NewDatabaseName $newDatabaseName -LogPath $bacpacImportLogPath -Verbose
              
if ($importBacpacResponse -ne $null) {
    Write-Host $importBacpacResponse
    Write-Host "$(Get-Date) - Import of bacpac file finished. New database name: $($newDatabaseName)"
} else {
    Write-Host "$(Get-Date) - Import of bacpac file failed"
}
```

I got the "failed" msg indicating the response obj is null even though the sqlpackage output indicates success.

Output:
```
...
Enabling index 'PK_JobId'.
Enabling index 'Idx_CreatedAtOrBeforeDateTimeUtc'.
Enabling index 'PK_PlanVersion'.
Successfully imported database.
Time elapsed 13:30:53.65
VERBOSE: [03:53:00][Invoke-Process] Total time spent inside the function was 13:31:49.7336837
VERBOSE: [03:53:00][Invoke-SqlPackage] Total time spent inside the function was 13:31:49.7516897
VERBOSE: [03:53:00][Import-D365Bacpac] Importing completed
VERBOSE: [03:53:00][Import-D365Bacpac] Start working on the configuring the new database
VERBOSE: [03:53:00][Import-D365Bacpac] Building sql statement to update the imported SQL database
VERBOSE: [03:53:00][Get-SQLCommand] Building the SQL connection string.
VERBOSE: [03:53:01][Import-D365Bacpac] Total time spent inside the function was 13:31:51.3040555
04/20/2023 03:53:01 - Import of bacpac file failed
```

I am puzzling what is the shortest way to check in code for the success of that command. Executing this command is so timetaking I thought it's better to ask. Is there some parameter I overlooked in order to get a response obj? Or do I have to inspect the log file?

Thanks in advance, Jakob

**Comment by Splaxi** [comment, marked as answer]
The command originally was developed with human invocation and interaction in mind.

So if the cmdlet doesn't fail - like hardcore throwing exceptions around and stuff like that - the import was successful 😉

**Comment by jakobschroeber** [comment]
Ok I will take your answer as 'there will never be a response object for Import-D365BacPac'. I guess I will proceed by checking if I catch an exception from that command, having set the -EnableException parameter to true

Thanks
